### ISSUE TITLE
maintaining comment order for @apiParam and @apiSuccess

### ISSUE THREAD

**Issue by nazrhyn** [apidoc/apidoc#206]
Currently, all tags inherited by the use of `@apiUse` end up, in the template, at the end of their table. Here's an example.

Let's say that an API always returns a consistent outer structure for results, like the following:

``` js
{
    "status": "STATUS_CODE",
    "data": {
        // result data
    }
}
```

In that case, in order to prevent having to write it over and over, my first inclination would be to create something like this:

``` js
/**
 * @apiDefine SuccessData
 * @apiSuccess {Object} data The response data object.
 */
```

_A similar definition might be created for an API key parameter which must be supplied with every request._

When documenting a resource, I might produce something like this:

``` js
/**
 * @api {get} /dog/:id Get Dog
 * @apiName GetDog
 * @apiGroup Dogs
 * 
 * @apiDescription Gets the dog identified by `id`.
 *
 * @apiParam {Number} id The dog ID.
 *
 * @apiUse SuccessData
 * @apiSuccess {String} data.name The name.
 * @apiSuccess {String} data.breed The breed.
 * @apiSuccess {Number} data.age The age, in human years.
 * @apiSuccessExample {json} Success Example
 * HTTP/1.1 200 OK
 * {
 *   "status": "OK",
 *   "data": {
 *     "name": "Fido",
 *     "breed": "Husky",
 *     "age": 6
 *   }
 * }
 */
```

When this is rendered, though, the `data` field does not maintain its comment order. It moves to the bottom of the table, which visually breaks the structure implied by the indentation of the **Field** column.

![apidoc-dog-example](https://cloud.githubusercontent.com/assets/1672405/6087484/04081134-ae1a-11e4-92fe-25a456083ec9.png)

Because of this effect, there are some things that I have had to duplicate rather than just inherit via `@apiUse`. Would it be possible to maintain the order as it is defined in the comment?

**Comment by mauler** [comment]
https://github.com/apidoc/apidoc/pull/457

**Comment by Yopai** [comment]
Got quite the same problem. Given the configuration :

    /**
     * @apiDefine apiResult
     * @apiSuccess {bool}   success   Whether the request has succeeded
     * @apiSuccess {Object} data
     */
    /**
     * @api        {get} /company Request company information
     * @apiName    GetCompany
     * @apiGroup   General
     *
     * @apiUse apiResult
     *
     * @apiSuccess {int}    data.id                      Company Id
     * @apiSuccess {string} data.name
     * @apiSuccess {string} data.company_code
     * @apiSuccess {int}    data.nb_user
     * @apiSuccess {date}   data.end_subscription_date
     * @apiSuccess {string} data.interface
     * @apiSuccess {string} data.params                 Semicolon-separated list of specific parameters
     * @apiSuccess {int}    data.quota_total            Total quota allocated (in bytes)
     * @apiSuccess {int}    data.quota_used             Total disk usage (in bytes)
     */

The result is

    success | bool | Whether the request has succeeded
    data | Object |  
        **quota_used | int | Total disk usage (in bytes)**
        id | int | Company Id
        name | string |  
        company_code | string |  
        nb_user | int |  
        end_subscription_date | date |  
        interface | string |  
        params | string | Semicolon-separated list of specific parameters
        quota_total | int | Total quota allocated (in bytes)

Why the last parameter (here, quota_used) is put above all the others ?

**Comment by klakekent** [comment]
still not fix this issue?

**Comment by theprometeus** [comment]
I was having this issue, it was introduced by the fix #457 that tried to fix the original issue #310.
I've made a pull request #951 refactoring the main.js/sortFields() function, and now it works as intended.

Before:
![image](https://user-images.githubusercontent.com/3390576/111403014-036df700-86ab-11eb-8a49-e5b3b0e164bf.png)

After:
![image](https://user-images.githubusercontent.com/3390576/111402697-732fb200-86aa-11eb-9723-a247976f0d92.png)
